### PR TITLE
Added continuousLine prop to line chart

### LINKED ISSUE #468: LineChart doesn't connect points that skip a point
### What is the problem?

The LineChart doesn't connect points when they skip 1+ point.

### When does it happen?

Render a LineChart with a missing point in the middle. Eg. When your XAxis is Monday-Sunday, a point for Tuesday is missing.

### What platform?

-   [X] iOS 
-   [ ] Android 

react-native version: `0.61`
react-native-svg-charts version: `5.3.0`
react-native-svg version: `9.13.3`

### Code to reproduce

```javascript
// put code here
```

### PR DESCRIPTION
**Pull request by rferdosi**
Added continuousLine props to line chart to skip null items in data.

fixes: #468,

### CODE REVIEW

File: src/line-chart/line-chart.js [hunk 1]
```diff
@@ -3,14 +3,22 @@ import Chart from '../chart/chart'
...
+                .y((d) => y(d.y))
+                .curve(curve)(data.filter((item) => item.y))
+        } else {
+            line = shape
+                .line()
+                .x((d) => x(d.x))
+                .y((d) => y(d.y))
+                .defined((item) => typeof item.y === 'number')
+                .curve(curve)(data)
+        }
```
Comment (JesperLekland) [inline src/line-chart/line-chart.js:21]: Wouldn't this work? 
Basically it exposes the `defined` function to user land, allowing them to implement it anyway they see fit 
```suggestion
        const { 
          defined = (item) => typeof item.y === 'number' ,
          continuousLine
          } = this.props
        const shape =
                .line()
                .x((d) => x(d.x))
                .y((d) => y(d.y))
                .defined(defined)
                .curve(curve)(data)
```

**Review by JesperLekland — CHANGES_REQUESTED** [review]
Thank you for your contribution! In this library I try to introduce as few "custom" props as possible, meaning that I instead try to expose the underlying building blocks, e.g `defined` so that the user can do more powerful things and the implementation stays clean. Creating a prop for each specific use case would very quickly bloat this library and make the implementation very hard to follow.

New properties also need to thoroughly be documented in the README.

File: src/line-chart/line-chart.js (hunk 1 above)
Comment (rferdosi) [reply src/line-chart/line-chart.js:21]: Hi, can you explain how can I use `defined` function for this situation?

Comment (JesperLekland) [reply src/line-chart/line-chart.js:21]: Hi, my suggestion above shows that `defined` is exposed as a prop. Meaning that we can have a default value (the one it is currently)  and that the user can override it how they see fit

Comment (rferdosi) [reply src/line-chart/line-chart.js:21]: I tried to implement the `defined` function for this specific use case but I failed. Do you have any idea?
That would help me to solve the main problem.

Comment (JesperLekland) [reply src/line-chart/line-chart.js:21]: If you could provide a PR with your attempt and clearly stating what isn't working I might be able to help you 👍

Comment (rferdosi) [reply src/line-chart/line-chart.js:21]: Can you help me with this?
Actually I think it's impossible to handle this use case without filtering data.
Checkout [this](https://observablehq.com/@d3/line-with-missing-data) example from d3.

Comment (JesperLekland) [reply src/line-chart/line-chart.js:21]: I unfortunately don't have any time to assist outside of reviewing the PR. The user controls the data so I don't see why the user won't be able to filter the data? What exactly are you referring to in your example?

**Comment by brianbscho** [comment]
Hi, I guess I'm late on the discussion but I also need this PR so want to add some comments.

Difference between `data.filter` (PR) and `defined` is that `data.filter` draws line between two dots where nulls in between but `defined` skips and makes empty space as noted in [d3-shape](https://github.com/d3/d3-shape#line_defined). Once it used `defined` already, there's no way to draw line. So exposing `defined` as prop doesn't solve problem that this PR suggests, which solves #266 #399 #468 . Can you please merge this PR?

File: src/line-chart/line-chart-grouped.js [hunk 2]
```diff
@@ -3,17 +3,27 @@ import ChartGrouped from '../chart/chart-grouped'
...
-        const lines = data.map((line) =>
-            shape
+        const { curve, continuousLine } = this.props
+        let line
+        if (continuousLine) {
+            line = shape
+                .line()
+                .x((d) => x(d.x))
+                .y((d) => y(d.y))
+                .curve(curve)(data.filter((item) => item.y))
```
Comment (brianbscho) [inline src/line-chart/line-chart-grouped.js:13]: I think this should look like below, since `data` is array of `line`.

```suggestion
            lines = data.map((line) =>
                shape
                .line()
                .x((d) => x(d.x))
                .y((d) => y(d.y))
                .curve(curve)(line.filter((item) => item.y))
```

File: src/line-chart/line-chart.js (hunk 1 above)
Comment (ALPNP) [reply src/line-chart/line-chart.js:21]: @JesperLekland For example, we want to show the week, and the data is only available for Monday and Thursday. Line Chart will not be able to connect Monday and Thursday.

**Comment by adelaadeoye** [comment]
@rferdosi and @JesperLekland Is this solution available for general usage, I just installed the package and still can't have a continuesLine

File: src/line-chart/line-chart.js (hunk 1 above)
Comment (zirho) [reply src/line-chart/line-chart.js:21]: it's not only ` .defined(defined)` but also the data being fed to the chain is different.
